tox.ini

```diff
@@ -1,39 +1,69 @@
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/pure-python
 [tox]
-envlist = coverage-clean,py27,py35,py36,py37,py38,pypy,pypy3,coverage-report
-# Note: if you add new Python versions, please add them to
-# [testenv:coverage-report] depends as well!
+minversion = 3.18
+envlist =
+    lint
+    py27
+    py35
+    py36
+    py37
+    py38
+    py39
+    pypy
+    pypy3
+    coverage
 
 [testenv]
-commands =
-    coverage run -m zope.testrunner --test-path=src --all {posargs:-v --auto-color}
-setenv =
-    COVERAGE_FILE=.coverage.{envname}
+usedevelop = true
 deps =
-    coverage
+    zope.testrunner
+commands =
+    zope-testrunner --test-path=src {posargs:-vc}
 extras =
     test
-# parallel mode: make sure coverage-clean finishes running before we run this one
-depends = coverage-clean
 
-[testenv:coverage-clean]
-deps = coverage
-setenv =
-    COVERAGE_FILE=.coverage
+[testenv:lint]
+basepython = python3
 skip_install = true
-commands = coverage erase
-# parallel mode: be sure we don't inherit depends from the base [testenv]
-depends =
+deps =
+    flake8
+    check-manifest
+    check-python-versions
+commands =
+    flake8 src setup.py
+    check-manifest
+    check-python-versions
 
-[testenv:coverage-report]
-deps = coverage
-setenv =
-    COVERAGE_FILE=.coverage
-skip_install = true
+[testenv:coverage]
+basepython = python3
+allowlist_externals =
+    mkdir
+deps =
+    coverage
+    coverage-python-version
+    zope.testrunner
 commands =
-    coverage combine
-    coverage report
+    mkdir -p {toxinidir}/parts/htmlcov
+    coverage run -m zope.testrunner --test-path=src {posargs:-vc}
     coverage html
-    coverage xml
-# parallel mode: make sure all builds complete before we run this one
-depends = py27,py35,py36,py37,py38,pypy,pypy3
-parallel_show_output = true
+    coverage report -m --fail-under=85
```

Comment:
Huh, only 85%?  Rewrite-if-statement-to-oneliner tricks were usually done to get to 100%, so maybe I was mistaken.

Reply:
According to https://coveralls.io/github/zopefoundation/zope.app.wsgi `master` has also just 84 % coverage.